pca_for_index/pca_analysis.Rmd

```diff
@@ -0,0 +1,439 @@
+dimdesc(result_pca, axes=c(1,2))
+dimdesc(result_pca, axes=c(3,4))
+dimdesc(result_pca, axes=c(5,6))
+
+# Compute variable loadings 
+var_loadings <- sweep(result_pca$var$coord,2,sqrt(result_pca$eig[1:ncol(result_pca$var$coord),1]),FUN="/")
+
+# Contributions of the variables to the principal components. Variable contributions = (var.cos2 * 100) / (total cos2 of the component). Recall: The contribution of a point to the inertia of an axis is the quotient between the inertia of its projection and the inertia of the whole scatterplot's projection on this axis. 
+head(result_pca$var$contrib)
+
+# Plot variables and the correlations. The color gradient indicates variable contribution 
+fviz_pca_var(result_pca, 
+             axes = c(1, 2),
+             col.var="contrib", 
+             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"),
+             repel = TRUE     # Avoid text overlapping
+             )
+
+fviz_pca_var(result_pca, 
+             axes = c(3, 4),
+             col.var="contrib", 
+             #repel = TRUE,    # Avoid text overlapping
+             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07")
+             )
+
+
+fviz_pca_var(result_pca, 
+             axes = c(5, 6),
+             col.var="contrib", 
+             #repel = TRUE,     # Avoid text overlapping
+             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07")
+             )
+```
+
+
+# Building the index
+
+We build the composite housing quality index as the weighted sum of the obtained dimension scores for each household (Sekhar, Indrayan, & Gupta, 1991; Antony & Visweswara Rao, 2007). Specifically, we compute a non-standardized composite index (NSCI) for observation $j$ using the PVE explained by each component, $i$, as weights: 
+
+$$
+\textrm{NSCI}_j = \sum_{i=1}^6 \frac{\textrm{PVE}_{\textrm{Comp i}}}{\textrm{Total PVC}} \times \textrm{Score}_{\textrm{Comp i}}
+$$
+
+As seen in Figure ___, the index follows a bimodal normal distribution. There are several outliers on the extreme right. 
+
+```{r}
+# Scores (i.e. principal coordinates) for each observation
+scores <- as.data.frame(result_pca$ind$coord)
+
+## Import data from SPSS
+total_pvc <- pve[6,3]
+hq_index <- ((pve[1,2]*scores[,1]) + (pve[2,2]*scores[,2]) + (pve[3,2]*scores[,3]) + (pve[4,2]*scores[,4]) + (pve[5,2]*scores[,5]) + (pve[6,2]*scores[,6]))/total_pvc
+
+
+hist(hq_index, breaks=100)
+summary(hq_index)
+
+length(which(hq_index>5))
+```
+
+
+We standardize the index (SI) to a range from 0 to 10 (lowest to highest housing quality), as follows:
+
+$SI_j = \frac{NSCI_j-Min(NSCI)}{Max(NSCI)-Min(NSCI)} * 10$
+
+```{r}
+std_hq_index <- (hq_index-min(hq_index)) / (max(hq_index)-min(hq_index)) * 10
+summary(std_hq_index)
+
+hist(std_hq_index, breaks=100)
+```
+
+
+```{r}
+# Create dataset that contains component scores and standardized index
+df_index <- scores %>% mutate(housing_qual_index = std_hq_index)
+```
+
+# Alternative way to build the index
```

Comment:
I like this alternative index a lot! I'd be happy to mention just this one in the presentation / paper.